apiserver/params/params.go

```diff
@@ -1276,3 +1276,17 @@ type DumpModelRequest struct {
 	Entities   []Entity `json:"entities"`
 	Simplified bool     `json:"simplified"`
 }
+
+type UpgradeSeriesStatusResult struct {
+	Error  *Error                  `json:"error,omitempty"`
+	Status UnitSeriesUpgradeStatus `json:"status,omitempty"`
```

Comment:
Coming in late, but what is the rationale for having a particular type here rather thatn just a string?

Since this is only wire protocol, different types give us no benefit.

Reply:
Also a bit weird returning the status in a set result call.

Reply:
Fixed, we just pass around base string now. With validations to convert it into a typed const.